lib/LaTeXML/Package/aas_support.sty.ltxml

```diff
@@ -296,6 +309,68 @@ RequirePackage('deluxetable');
 Let(T_CS('\begin{planotable}'), T_CS('\begin{deluxetable}'));
 Let(T_CS('\end{planotable}'),   T_CS('\end{deluxetable}'));
 
+DefConditional('\ifcolnumberson');
+DefConditional('\ifdeluxedecimals');
+DefMacro('\deluxedecimals', '\global\deluxedecimalstrue');
+RawTeX('\global\deluxedecimalsfalse');
+Let(T_CS('\decimals'), T_CS('\deluxedecimals'));
+DefMacro('\colnumbers',              '');                                       # TODO
+DefMacro('\deluxedecimalcolnumbers', '\deluxedecimalstrue\colnumbersontrue');
+Let(T_CS('\decimalcolnumbers'), T_CS('\deluxedecimalcolnumbers'));
+
+# http://tug.ctan.org/tex-archive/macros/latex/contrib/aastex/sample63.pdf
+# decimal alignment, 3.1.2
+DefColumnType('D', sub {
+    $LaTeXML::BUILD_TEMPLATE->addColumn(
+      before => Tokens(T_CS('\hfill')),
+      after  => Tokens(T_CS('\end@D@col@one')));
+    $LaTeXML::BUILD_TEMPLATE->addColumn(
+      after => Tokens(T_CS('\hfill')));
+    return; });
+
+DefColumnType('d', sub {    #TODO: these should be "eaten away" I think?
+    $LaTeXML::BUILD_TEMPLATE->addColumn(
+      before => Tokens(T_CS('\start@D'), T_CS('\hfill')),
+      after  => Tokens(T_CS('\end@D@col@one')));
+    $LaTeXML::BUILD_TEMPLATE->addColumn(
+      after => Tokens(T_CS('\hfill')));
+    return; });
+
+DefMacro('\end@D@col@one', sub {
+    my ($gullet)         = @_;
+    my $smuggle_new      = 1;
+    my $current_location = $gullet->getLocator->toString;
+    if (my $smuggled_at = LookupValue('aas_smuggle_D_column')) {
+      if ($smuggled_at eq $current_location) {
+        $smuggle_new = 0;
+    } }
+    if ($smuggle_new) {
+      AssignValue('aas_smuggle_D_column', $current_location, 'local');
+      return T_CS('\@alignment@align'); }
```

Comment:
I only arrived at this locator-based solution for adding an extra alignment in the table body "by the skin of my teeth".

It's a very ugly problem, because the `D` spec has a split personality. 
 - Face 1: In the table header, authors are explicitly requested to use a `\multicolumn2c` specifier for the `D` entries, to fill two entries.
 - Face 2: In the table body, authors are expected to just casually deposit their decimal numbers in a single column and the magic of `\lookfordecimal` creates a second alignment behind the scenes.

I did not emulate the decimal alignment yet, but I managed to get the "second align behind the scenes" behavior, using this very awkward locator trick. What happens if this guard isn't added? An infinite loop apparently, since the newly deposited alignment causes a new `D` first column to be created, which drops another new alignment, and so on into infinity.

So the locator assignment makes sure the expansion doesn't create a universe of "align tokens all the way down".

Reply:
It would of course be easier if, as in `numprint.sty.ltxml`, the `D` column type only ever created one column. But then I would have to "detect" an extra column in the header and remove its alignment. I guess that's also doable, and may be an alternative avenue... Feedback most welcome.

Reply:
And, yeah; A very scary version of what I knew would be required with "D" creating two columns.  I'll have to ponder...

However,is "split personality" a perverse pun?  It isn't so much itself split: It creates two columns, but has a before/after that causes it to read only up to one "&" but split it!  The `\multicolumn` is just there to cause the column heading to span 2 columns.  Of course the over cleverness of `Alignment` causes (in this example) the extra, empty, column (and the span) to disappear --- this threw me off while testing!

Reply:
Incidentally: what you've written is in the ballpark of correct. It's only a question of do you *want* it to create 2 columns split on the decimal (and thus hide the real number for posterity), or just ignore the alignment (until html catches up)?

Reply:
"ballpark of correct" was hard enough!

Maybe I should've gone with "Two-face" instead of split personality, if you prefer Batman references :> The backend is certainly unified (oh, it's just two columns, snooze, boring), but the *frontend* is dual - the author is expected to *write* two columns in the header, and a single column in the body, even if the single column in the body will eventually "become" two.

And calling `->addColumn` twice in a single `DefColumnType` was something I questioned the entire time, since it feels like a very suspicious technique...

I think I prefer having "two columns" everywhere for now, to stay close to the TeX implementation, and also since I feel it is more predictable to add an alignment token in the body than to remove one in the header (from a \multicolumn command.. yikes.)

Reply:
By deciding on 2 columns, I meant whether the number would be split into two columns, or just add an empty column.  Actually, a more TeX-like approach is to provde a 'reader' for the begin of the first column, and a marker at it's end (You've got a thing about smuggling, perhaps?)  The following code seems to work, and aligns as desired:
```
DefColumnType('D', sub {
    $LaTeXML::BUILD_TEMPLATE->addColumn(
      before => Tokens(T_CS('\hfill'),T_CS('\aas@start@D@column')),
      after  => Tokens(T_CS('\aas@end@D@column')));
    $LaTeXML::BUILD_TEMPLATE->addBetweenColumn();
    $LaTeXML::BUILD_TEMPLATE->addColumn(
      after => Tokens(T_CS('\hfill')));
    return; });

DefMacro('\aas@start@D@column XUntil:\aas@end@D@column', sub {
  my($gullet,$n)=@_;
  my($m,$f)=SplitTokens($n,T_OTHER('.'));
  return ($f ? (@$m,
                T_CS('\@ADDCLASS'),T_BEGIN,T_OTHER('ltx_norightpad'),T_END,
                T_CS('\@alignment@align'),
                T_CS('\@ADDCLASS'),T_BEGIN,T_OTHER('ltx_noleftpad'),T_END,
                T_OTHER('.'),@$f)
          : ($n,T_CS('\@alignment@align')));
##  return ($n,T_CS('\@alignment@align'));
  });
DefPrimitive('\aas@end@D@column','');
```
The between column bit was an experiment that didn't work, yet. The idea is to suppress intercolumn spacing.  The temporary hack is to add the CSS class, which needs the following added to LaTeXML.css:
```
.ltx_norightpad { padding-right:0!important; }
.ltx_noleftpad { padding-left:0!important; }
```
Maybe that's what the between columns bit should be doing?

Reply:
Very nice! I am unsure what the magic of the `addBetweenColumn` does there, maybe understanding that would have saved me some pain :> Definitely not overly fond of smuggling in general, just that it keeps working reliably as a low-level technique in weird TeX circumstances...

Reply:
I've commited your changes to the branch since they're a clear upgrade btw.

I am now slowly pondering your code and trying to figure out why it doesn't fall into an infinite loop when it introduces the new `\@alignment@align`, while mine did, hm.

Reply:
It's likely the combination of your use of `XUntil` and the timing of `after=>` that avoids the loop. I can't quite wrap my mind around tracing the individual steps, but I'll keep in mind this type of setup results in cleaner code. Thanks a lot for the upgrade!!